### PR TITLE
HTSP: Add SNR and Signal status for Decibel scale.

### PR DESCRIPTION
**Pull request by bluzee**
Bump version to 37.

### CODE REVIEW

**Comment by Flole998** [comment]
That looks better. I am wondering though: What datatype is `sig->snr`? Shouldn't it rather be a float?

File: src/htsp_server.c [hunk 1]
```diff
@@ -4371,8 +4371,12 @@ htsp_subscription_signal_status(htsp_subscription_t *hs, signal_status_t *sig)
     htsmsg_add_str(m, "feStatus",   sig->status_text);
     if((sig->snr != -2) && (sig->snr_scale == SIGNAL_STATUS_SCALE_RELATIVE))
       htsmsg_add_u32(m, "feSNR",    sig->snr);
+    if((sig->snr != -2) && (sig->snr_scale == SIGNAL_STATUS_SCALE_DECIBEL))
+      htsmsg_add_u32(m, "feSNRDb",    sig->snr);
```
[the remark this answers is not in the document]
Comment (Flole998) [reply src/htsp_server.c:4375]: I believe those blanks were to align those values so it's easier to read, the line below is kinda breaking the formatting though.

**Comment by ksooo** [comment]
> multiplied by 10

almost, multiplied by 1000

```c++
int signal; ///< signal strength, value depending on signal_scale value:
              ///<  - SCALE_RELATIVE : 0...65535 (which means 0%...100%)
              ///<  - SCALE DECIBEL  : 0.0001 dBm units. This value is generally negative.
  int snr;    ///< signal to noise ratio, value depending on snr_scale value:
              ///<  - SCALE_RELATIVE : 0...65535 (which means 0%...100%)
              ///<  - SCALE DECIBEL  : 0.0001 dB units.
```

**Comment by bluzee** [comment]
I changed u32 to s32 for both SNR and Signal.  Signal will always be negative dbm.   Think I found all the blank spaces.

**Comment by 7687476874** [comment]
I would be very grateful if you could let me know if the signal indicator will work on kodi or not.

**Comment by ksooo** [comment]
> I would be very grateful if you could let me know if the signal indicator will work on kodi or not.

This will not work out of the box. After this PR here went in, Kodi needs to be changed to support the dB/dBm values. Unfortunately, this is more work than you might think. Its just displaying two strings, right? But for this Kodi itself and all PVR add-ons need to be touched as this new feature requires a Kodi PVR add-on API change, even if in a first step only pvr.hts would support the new values. This is how things work with Kodi add-on API changes.

So, you will need very latest tvheadend, Kodi and pvr.hts. On the Kodi side nothing that will be available before Kodi 20. As you can see, high effort, just for some (imo not important) strings in Kodi user interface. Somebody needs to do the work on Kodi side. If nobody steps up, this feature will never happen. That's how it goes with open source projects run by 100% volunteers.

BTW: What is the whole point about these values? Once you setup your hardware (e.g. sattellite dish) correctly, SNR and signal strength should be stable and there is not need to look at them on a every time you watch Live TV. Look into the tvheadeend GUI if you feel there is something wrong with SNR/signal strength. ;-)

**Comment by 7687476874** [comment]
Thank you very much for your information.

**Comment by bluzee** [comment]
Changing the whole API for this seems a bit more trouble than it's worth. TVH is the only backend that is fully functional for live TV so nothing else will make use of this.  HTSP is only for TVH and pvr.hts correct? TVH web GUI used the API instead correct?  What would happen if we just did this in HTSP....

   if((sig->snr != -2) && (sig->snr_scale == SIGNAL_STATUS_SCALE_RELATIVE))
      htsmsg_add_u32(m, "feSNR",    sig->snr);
    if((sig->snr != -2) && (sig->snr_scale == SIGNAL_STATUS_SCALE_DECIBEL))
      htsmsg_add_s32(m, "feSNR",    sig->snr);

It's only going to ever send feSNR once correct?  Is the string sent by SCALE_DECIBEL something the existing pvr.hts can deal with?  

Satellite dishes  rotate across the arc. TV antennas rotate to point to different transmitter locations.  They often need nudging. One pulse can fix a signal. 

If I force old stats I get more or less meaningless stats in both Kodi and TVH gui. If I use dB I get useful numbers in TVH gui but currently 0 in Kodi.  I don't really care if Kodi shows a % sign or says dB.  If it sends feSNR 13500 I will know that is 13.5dB and that to me is a meaningful number.  So if we can get a number into Kodi without having to rewrite the whole thing that would be wonderful.  Rewriting every PVR addon would be pointless as TVH is the only backend that is suitable  for this.

**Comment by 7687476874** [comment]
I found an image in tvheadend version 4.2

Screenshot_20220513_202557

**Comment by 7687476874** [comment]
I found an image in tvheadend version 4.2
Kodi18 1_SNR_0percent_screenshot000

**Comment by ksooo** [comment]
> Changing the whole API for this seems a bit more trouble than it's worth.

 @bluzee I'm sick to explain things again and again. If you know better, come up with some real code. Open respective PRs for Kodi/pvr.hts. Then we have something real to discuss. Personally, I couldn't care less for this feature. Given my limited time for Kodi, this has very low priority on my long list of things to do.

**Comment by bluzee** [comment]
OK.  I'll figure it out on my own. In future if you have absolutely no interest in something just come out and say it. When the request came in on Kodi forum you dumped it on TVHeadend. You told people that you knew exactly how to fix the issue if only TVHeadend would get their act together. There is nothing wrong with simply stating that you have no interest in something.  This would have stopped then.  Anyway it's stopping now.

**Comment by ksooo** [comment]
> You told people that you knew exactly how to fix the issue

 @bluzee  Yes, I do. Now, you started questioning what I proposed and this is the point I'm out.

**Comment by ksooo** [comment]
BTW, you should not close this PR as THIS is exactly the right way to go, the first step in the chain of changes needed.

**Pull request closed without merging**